src/mainAppBar/mainAppBar.component.tsx

```diff
@@ -252,6 +253,9 @@ const mapStateToProps = (state: StateType): MainAppProps => ({
   showAdminPageButton:
     state.scigateway.authorisation.provider.isLoggedIn() &&
     state.scigateway.authorisation.provider.isAdmin(),
+  showUserComponent:
+    state.scigateway.authorisation.provider.constructor.name !==
```

Comment:
Could we use `instanceof` instead here? `state.scigateway.authorisation.provider instanceof NullAuthProvider`? It's less fragile